### ISSUE TITLE
ERROR: MariaDB doesn't yet support 'CACHE without INCREMENT BY 0 in Galera cluster'

### ISSUE THREAD

**Issue by beckerr-rzht** [privacyidea/privacyidea#4012]
### Top-level intent

Clone database running on MariaDB 10.11 + Galera 26.4.18

### Steps to reproduce

```bash
echo "create database clone;" | mysql -BN
mysqldump --single-transaction --add-drop-table --ignore-table-data pi.pidea_audit pi | mysql -BN clone
```

### Expected outcome

A clone of database `pi` named `clone`.

### Actual outcome

Error Message:
```
ERROR 1235 (42000) at line 248: This version of MariaDB doesn't yet support 'CACHE without INCREMENT BY 0 in Galera cluster'
```
The message caused by this SQL statement:
```
CREATE SEQUENCE `serviceid_seq` start with 1 minvalue 1 maxvalue 9223372036854775806 increment by 1 cache 1000 nocycle ENGINE=InnoDB
```
In particular by `... increment by 1 cache 1000 ...`

But all other sequences are defined in this way:
```text
.. increment by 0 cache 1000 ..
```
Why is it different for serviceid_seq and can it be changed to make privacyIDEA compatible with MariaDB 10.11 + Galera 26.4.18?

### Configuration

* **privacyIDEA version**: 3.9.3
* **Installation method**: venv pip
* **Python version**: 3.12
* **Operating system**: Ubuntu 24.04 LTS
* **Webserver**: Apache2
* **Token database**: MariaDB 10.11 + Galera 26.4.18

**Comment by beckerr-rzht** [comment]
Could it be that `service_seq` is a legacy thing that could simply be removed?

The sequence does not even appear in the following list: [5cb310101a1f_.py#L40](https://github.com/privacyidea/privacyidea/blob/e5e0027d61bda50fb0c2e8621fb995fe387085e7/migrations/versions/5cb310101a1f_.py#L40)

**Comment by plettich** [comment]
Hi @beckerr-rzht thanks for the report, we'll look into it.
All the sequences should have been adjusted to work with galera but it looks like this one slipped through.

**Comment by beckerr-rzht** [comment]
Yes, there are big differences between PostgreSQL and Mariadb/Mysql as far as sequences or auto increment are concerned.

Since privacyIDEA uses sequence table, however, I have repeatedly observed primary key collision, as I have described in #3810. In my experience, it is very error-prone to operate privacyIDEA on two nodes at the same time. Collisions occur almost immediately in `pidea_audit` table.
